src/main/java/org/openjdk/jextract/JextractTool.java

```diff
@@ -625,6 +601,15 @@ private Optional<Path> inferMacOsPath(String... paths) {
     }
 
     private String formatFrameworkPath(String optionString) {
-        return String.format(":/System/Library/Frameworks/%1$s.framework/%1$s", optionString);
+        String publicPath = String.format("/System/Library/Frameworks/%1$s.framework/", optionString);
+        String privatePath = String.format("/System/Library/PrivateFrameworks/%1$s.framework/", optionString);
```

Comment:
I think Maurizios comment still applies. AFAIU we need to pass the exact framework path to `dlopen`, so we need to manually search for the frameworks in this system location, and also in the locations specified through `-F`, and then load it from one of those locations.